pkg/job/job_test.go

```diff
@@ -0,0 +1,482 @@
+package job_test
+
+import (
+	"fmt"
+	"math/rand"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/MinhNghiaD/jobworker/api/worker/proto"
+	"github.com/MinhNghiaD/jobworker/pkg/job"
+	"github.com/sirupsen/logrus"
+)
+
+func TestStartJobs(t *testing.T) {
+	logrus.SetOutput(os.Stderr)
+	manager, err := job.NewManager()
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	defer func() {
+		if err := manager.Cleanup(); err != nil {
+			t.Fatal(err)
+		}
+	}()
+
+	// checker
+	checkStartCmd := func(cmd string, args []string) error {
+		jobID, err := manager.CreateJob(cmd, args, "test user")
+
+		if err != nil {
+			return err
+		}
+
+		if j, ok := manager.GetJob(jobID); ok {
+			defer j.Stop(false)
+		}
+
+		return nil
+	}
+
+	// TODO: add more test cases and reinforce the behaviour of the command execution.
+	// NOTE: For now, we accept that user can start job with high privilege. The job will be started but the execution will fail.
+	testcases := []struct {
+		name      string
+		cmd       string
+		args      []string
+		expectErr bool
```

Comment:
I would prefer to expect the error type rather than whether or not there is any error

Reply:
Yes, I think that could be better. However, some of the returned errors come from internal of the go package and they are various. Therefore I just check how my code handled these errors and the behaviors are as expected or not

Reply:
Hi @nklaassen I have added the error code cross-check as you suggested and it's beautiful. Thank you very much